### DISCUSSION TITLE
AZ_* environment variables missing

### DISCUSSION THREAD

**Discussion by bytepl** [JanDeDobbeleer/oh-my-posh#984] [Troubleshoot]
Hi,

I am trying to add Azure segment to my prompt but it's missing, whatever I do. I added a segment to the theme but it does not seem to do anything:

{
          "type": "az",
          "style": "powerline",
          "powerline_symbol": "\uE0B0",
          "foreground": "#000000",
          "background": "#9ec3f0",
          "properties": {
            "display_id": true,
            "display_name": true,
            "info_separator": " @ ",
            "prefix": " \uFD03 "
          }

Write-PoshDebug shows that the segment evaluates to false, at the same time Get-AzContext shows that I am connected.

I have noticed that I am missing AZ_* variables in my session, the only one present is AZ_ENABLED (set to 'True'). I tried adding any of the following manually but nothing happens, these just not show up:

AZ_ENVIRONMENT_NAME
AZ_USER_NAME
AZ_SUBSCRIPTION_ID
AZ_ACCOUNT_NAME

Should these variables be set up by Connect-AzAccount?

Looks like oh-my-posh is looking for these variables:

2021/09/22 11:50:40 getenv duration: 0s, args: AZ_ENVIRONMENT_NAME
2021/09/22 11:50:40 getenv duration: 0s, args: AZ_USER_NAME
2021/09/22 11:50:40 getenv duration: 0s, args: AZ_SUBSCRIPTION_ID
2021/09/22 11:50:40 getenv duration: 0s, args: AZ_ACCOUNT_NAME

**Comment by JanDeDobbeleer** [comment]
You need to either have the az command and authenticated OR the Powershell module available and authenticated with the env var set as you did.

**Reply by bytepl** [reply]
I do not have az installed, I have Azure Powershell installed and I am authenticated in my subscription, Get-AzContext returns my Azure context.

When I try to do this:

$context = Get-AzContext
$env:AZ_ENVIRONMENT_NAME = $context.Environment.Name
$env:AZ_USER_NAME = $context.Account.Id
$env:AZ_SUBSCRIPTION_ID = $context.Subscription.Id
$env:AZ_ACCOUNT_NAME = $context.Subscription.Name

the variables are still missing. That's the weird part :)

**Reply by JanDeDobbeleer** [reply]
I'll check!

**Reply by JanDeDobbeleer** [reply]
@bytepl this works on my machine BUT I noticed that I was no longer authenticated so had to run ` Connect-AzAccount -DeviceCode` first to set up the pwsh AZ module.

**Reply by JanDeDobbeleer** [reply]
@bytepl I did find a mistake in a recent refactor, can you try again with the next version?

**Reply by bytepl** [reply]
@JanDeDobbeleer Works just fine, thank you so much!
